### PR TITLE
MRCPv2 support

### PR DESCRIPTION
**Pull request by MayamaTakeshi**
Hi, 
I have implemented MRCPv2 support.  
I read https://github.com/irontec/sngrep/issues/254 but didn't like the approach of treating MRCP messages as an rtp_stream.
Instead I adjusted MRCP messages to be handled as SIP messages and based on the channel-identifier returned by the server in its SDP, to add the MRCP messages to the corresponding SIP call.
I have tested and could not find problems using both the offline mode and the live mode with thousands of calls.

(but there are some things that don't work yet like not being able to save the MRCP messages to file).

I'm attaching a sample capture file with MRCP with one session for Speech Synthesis and another for Speech Recognition in case you need them.
[mrcp.pcap.gz](https://github.com/irontec/sngrep/files/5982013/mrcp.pcap.gz)

### CODE REVIEW

File: src/sip.c [hunk 1]
```diff
@@ -354,15 +423,27 @@ sip_check_packet(packet_t *packet)
     // Get Method and request for the following checks
     // There is no need to parse all payload at this point
     // If no response or request code is found, this is not a SIP message
-    if (!sip_get_msg_reqresp(msg, payload)) {
+    if(channelid[0]) {
+        if(!sip_get_msg_reqresp_for_mrcp(msg, payload)) {
+            printf("bad bad\n");
+            exit(1);
```
Comment (Kaian) [inline src/sip.c:429]: I think this is debugging code, you should not exit sngrep from capture layer.

**Review by Kaian — COMMENTED** [review]
Hi @MayamaTakeshi 

Thanks a lot for taking the time to understand the code and implement this. It's awesome and also fits with existing code style.

Sadly, current version of sngrep is only receiving bugfixes so I will prefer to add this feature to a future version. 

If you check glib-2 branch you'll see we have been reworking all sngrep capture layer and split each protocol logic to its own packet dissector. This is a better and cleaner implementation that allows users to configure with protocols are interested in and will support extending the capture layer with new protocols (like MRCP) with ease.

This PR can serve for those that need to have this feature in a forked sngrep 1.X and can not wait for an alpha release of the next major version, while I will adapt this changes to fit in a new packet dissector for 2.x

Thanks a lot for your hard work!

File: src/sip.c (hunk 1 above)
Comment (MayamaTakeshi) [reply src/sip.c:429]: Yes. Sorry about that.

**Comment by MayamaTakeshi** [comment]
Hi @Kaian,
no problem. 
It will be nice to see an architectural change that will simplify adding more protocol handlers.

And actually it was not much work as sngrep code is very easy to follow.
I did it in about 2 days as I've been working with MRCP a lot in the past months and will need this for debugging an app integration with unimrcp.

So if you get busy when the glib-2 branch becomes main, I might take care of adding MRCP support to it (to be frank I think there is not much people interested in it these days or at least there is no much need to look at packets to solve/investigate issues).

Thanks for this nice tool.

**Comment by Kaian** [comment]
Hi @MayamaTakeshi 

Above commit contains the adaptation of your Pull Request for 2.x branch, including the MRCPv2 Protocol dissector and MRCP channels hash table in storage. Feel free to check the code in case you have any comment. 

Take into account that if you want to try MRCP in 2.x you must first enable TCP and MRCP protocols (both are disabled by default) by adding this lines to your ~/.sngreprc file

     set packet.tcp.enabled on
     set packet.mrcp.enabled on

Any feedback is highly appreciated.

Thanks a lot for your contributions!

**Comment by Kaian** [comment]
Now I see some fixes like 

https://github.com/MayamaTakeshi/sngrep/commit/47ee7d61a64d104bb122181a9676cee91c4497b6 

Not sure if there are others, I'll try to also add them to glib-2 branch.

Regards!

**Pull request closed without merging**